MoveAndRename.md

```diff
@@ -0,0 +1,169 @@
+  on the same file system, or only for `rename()`d handles?
+
+# Handling Moves to Other Directories
+
+There are three cases to consider when moving to other directories:
+
+## (1) Moving to a Non-Local File System
+
+Moving across file systems cannot be guaranteed to be atomic if the operation
+cannot be completed successfully (e.g. network connection issues, lack disk
+space, etc.). This may result in partial writes to the target directory.
+
+Note that this approach is no worse than what is available currently.
+
+## (2) Moving out of the OPFS
+
+[#310](https://github.com/WICG/file-system-access/pull/310) exposes a strong use
+case for fast, atomic moves to other directories on the same file system. Files
+can be written efficiently in the Origin Private File System using an
+`AccessHandle`, then efficiently moved to a directory of the user's choice.
+However, writes using an `AccessHandle` are not subjected to Safe Browsing checks
+in Chrome. Moving files out of the OPFS to a user's directory will require
+running Safe Browsing checks on all moved files.
+
+For example, moving a large directory across the OPFS -> file system boundary
+means locking and performing checks (Safe Browsing in Chrome) for an unbounded
+number of files.
+
+## (3) Moving to a Directory on the Local File System
+
+This is the most straightforward case. Every `rename()` falls in this case. The
+operation should be atomic and fast, since no Safe Browsing checks need to be
+performed. However, providing these guarantees for `move()` in only this case
+may be confusing.
+
+## Open Questions - Move
+
+### Should behavior be different when moving a file vs. moving a directory?
+
+- For a file:
+  - Atomicity is only guaranteed in (2) and (3), though is likely true for (1).
+  - Speed is only guaranteed in (3).
+- For a directory:
+  - Atomicity is only easily guaranteed in (3). Atomicity for (2) would be
+    complex (likely requiring locking each file as an unbounded number of files
+    are scanned) and may not be the best option.
+  - Speed is only guaranteed in (3), but the implications of this for (1) and
+    (2) is much greater than when only moving a file.
+
+### What should happen if the operation cannot be completed successfully?
+
+- Abort the operation
+  - Least complexity for browser implementers, meaning less room for
+    implementation-specific behavior.
+  - Simplest interface.
+  - Potential for errors not caught in development.
+  - Apps need code to handle the cross-filesystem case.
+- Abort the operation and attempt to remove partial writes
+  - This may not be possible. Ex: loss of network connection.
+  - Adds lots of complexity.
+- Abort the operation and throw an error
+  - Less complexity.
+  - No performance cliff.
+  - More flexibility in handling errors for the app.
+
+# Implementation Notes
+
+- Write permission needs to have been granted to both the source and destination
```

Comment:
To double-check: This means that the `rename` operation will require write permissions on the containing directory; meaning that giving write permission to file F or directory D using the file chooser permission UX will never allow renaming F or D, right?

Reply:
Correct. Renaming a file or directory will require write permission to its parent directory.

Reply:
Sounds good! Let's spell this out in the explainer!

Reply:
Having said that, I think we could consider an extension of this where giving write permission to files F and G using the file chooser permission UX will allow renaming from F to G as well (well, not sure if write permission on the source file itself should be enough, I guess that depends on if "self deletion" is something that we want to allow or not). But it seems that write permission on the destination file as opposed to the destination folder should conceptually be enough to allow a move to that file.

Reply:
It sounds reasonable to me to define the permission requirements for moving /dir1/file1 to /dir2/file2 as:
 a) Write permission on directory /dir1,
AND
 b) Write permission on directory /dir2 or write permission on file /dir2/file2, making sure to correctly apply MotW & co.

Reply:
Updated the doc to allow for write permission on the destination file, as described above.